modules/database/src/ioc/dbStatic/dbLexRoutines.c

```diff
@@ -1034,22 +1050,49 @@ static void dbBreakBody(void)
     }
     pgphentry->userPvt = pnewbrkTable;
 }
-
+
+static
+int dbRecordNameValidate(const char *name)
+{
+    size_t i=0u;
+    const char *pos = name;
+
+    if (!*name) {
+        yyerrorAbort("Error: Record/Alias name can't be empty");
+        return 1;
+    }
+
+    for(; *pos; i++, pos++) {
+        char c = *pos;
+        if(i==0) {
+            /* first character restrictions */
+            if(c=='-' || c=='+' || c=='[' || c=='{') {
+                errlogPrintf("Warning: Record/Alias name '%s' should not begin with '%c'\n", name, c);
+            }
+        }
+        /* any character restrictions */
+        if(c < ' ') {
+            errlogPrintf("Warning: Record/Alias name '%s' should not contain non-printable 0x%02u\n",
+                         name, (unsigned)c);
+
+        } else if(c==' ' || c=='\t' || c=='"' || c=='\'' || c=='.' || c=='$') {
```

Comment:
There is sime redundancy here. `c=='\t'` is already covered by `c < ' '` above (for any encoding? ASCII at least) and `c==' '` could have have been included abore as 'c <= ' '`.
If we need to handle `\t` separately, the what about `\r`, `\l` ... ?